rmltraintfbbox/rmltraintfbbox/core.py

```diff
@@ -119,78 +125,145 @@ def train(ctx: click.Context, train: TrainInput):
+
+    if callable(learning_rate):
+        learning_rate_fn = learning_rate
+    else:
+        learning_rate_fn = lambda: learning_rate
+
+    # restore from checkpoint
+    load_fine_tune_checkpoint(detection_model, train_config.fine_tune_checkpoint,
+                                    train_config.fine_tune_checkpoint_type,
+                                    train_config.fine_tune_checkpoint_version,
+                                    train_input,
+                                    train_config.unpad_groundtruth_tensors)
+
+    @tf.function
+    def train_step(detection_model, train_input_iterator, optimizer, learning_rate_fn, global_step):
+
+        features, labels = train_input_iterator.next()
+        loss = model_lib_v2.eager_train_step(detection_model, features,
+                            labels, train_config.unpad_groundtruth_tensors,
+                            optimizer, learning_rate_fn(),
+                            add_regularization_loss=True,
+                            clip_gradients_value=None,
+                            global_step=global_step)
+        global_step.assign_add(1)
+        return loss
+
+    # @tf.function
+    def evaluate(detection_model, configs, eval_input, global_step):
+
+        text_trap = io.StringIO()
+        sys.stdout = text_trap
+        sys.stderr = text_trap
+
+        metrics = model_lib_v2.eager_eval_loop(detection_model, configs, eval_input, global_step=global_step)
+
+        sys.stdout = sys.__stdout__
+        sys.stderr = sys.__stderr__
+
+        click.echo(f'Evaluation loss: {metrics["Loss/total_loss"]}, Evaluation mAP: {metrics["DetectionBoxes_Precision/mAP"]}')
+
+        return metrics
+
+    checkpoint = tf.train.Checkpoint(optimizer=optimizer, model=detection_model)
+    manager = tf.train.CheckpointManager(checkpoint, directory=model_dir, max_to_keep=5)
+
     with ExitStack() as stack:
         if comet:
             stack.enter_context(experiment.train())
         click.echo('Training model...')
-        tf.estimator.train_and_evaluate(estimator, train_spec, eval_specs[0])
-        click.echo('Training complete')
+
+        start = time.time()
+        # main training loop
+        for step in range(1, num_train_steps+1):
+
+            loss = train_step(detection_model, train_input_iterator, optimizer, learning_rate_fn, global_step)
+            
+            if step % config.get('log_train_every') == 0:
+                print(f'Training loss at step {step}: {loss}')
+                if comet:
+                    experiment.log_metric('loss', loss)
+                
+            if step % config.get('log_eval_every') == 0:
+                stack.enter_context(experiment.validate())
+                manager.save()
+                eval_metrics = evaluate(detection_model, configs, eval_input, global_step)
+                if comet:
+                    experiment.log_metrics(eval_metrics, step=step)
+                stack.enter_context(experiment.train())
```

Comment:
Anything with `experiment` should go inside an `if comet:` block. Also, should probably add averaging for the train loss like we discussed.